### PR TITLE
debug handler maintain through decomposition

### PR DESCRIPTION
**Pull request by Gasoonjia**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141612

Add checks in the ao numberic debugger to guard the debug handle consistency between aten op decomposition

Differential Revision: [D66517480](https://our.internmc.facebook.com/intern/diff/D66517480/)

### `torch.ao.quantization.pt2e.graph_utils.get_control_flow_submodules` is No Longer a Public API

We aim to make all functions under `torch.ao.quantization.pt2e.graph_utils` private. This update marks `get_control_flow_submodules` as a private API.

Currently, only `bfs_trace_with_node_process` remains public, but it will be marked private in the next release.

### `torch.ao.quantization.pt2e.graph_utils.get_control_flow_submodules` is No Longer a Public API

We aim to make all functions under `torch.ao.quantization.pt2e.graph_utils` private. This update marks `get_control_flow_submodules` as a private API. If you have to or want to continue using `get_control_flow_submodules`, please make a private call by using `_get_control_flow_submodules`.

Currently, only `bfs_trace_with_node_process` remains public, but it will be marked private in the next release.

**Example:**
Version 2.6:
```python
>>> from torch.ao.quantization.pt2e.graph_utils import get_control_flow_submodules
  ```

Version 2.7:
```python
>>> from torch.ao.quantization.pt2e.graph_utils import get_control_flow_submodules
ImportError: cannot import name 'get_control_flow_submodules' from 'torch.ao.quantization.pt2e.graph_utils'
>>> from torch.ao.quantization.pt2e.graph_utils import _get_control_flow_submodules # Note: Use _get_control_flow_submodules for private access
```

### CODE REVIEW

**Comment by Gasoonjia** [comment]
#suppress-bc-linter

File: test/quantization/pt2e/test_numeric_debugger.py [hunk 1]
```diff
@@ -128,29 +160,67 @@ def test_re_export_preserve_handle(self):
...
+
+            self._assert_each_node_has_debug_handle(ep)
+            pre_decomp_to_debug_handle_map_ref = (
+                self._extract_debug_handles_with_prev_decomp_op(ep)
+            )
+
+            ep_copy = copy.copy(ep)
+            ep_copy = ep_copy.run_decompositions()
+            self._assert_each_node_has_debug_handle(ep_copy)
+            pre_decomp_to_debug_handle_map = (
```
Comment (jerryzh168) [inline test/quantization/pt2e/test_numeric_debugger.py:215]: you mean `post`?

**Review by jerryzh168 — APPROVED** [review]

File: test/quantization/pt2e/test_numeric_debugger.py (hunk 1 above)
Comment (Gasoonjia) [reply test/quantization/pt2e/test_numeric_debugger.py:215]: It is `pre`. The map key is the pre-decompose operator name, and value is the debug handle in the post debug handle. Maybe I need to have a better name haha

Comment (jerryzh168) [reply test/quantization/pt2e/test_numeric_debugger.py:215]: oh I see, makes sense

File: docs/source/conf.py [hunk 2]
```diff
@@ -305,6 +305,7 @@
     "node_arg_is_weight",
     "return_arg_list",
     # torch.ao.quantization.pt2e.graph_utils
+    "bfs_trace_with_node_process",
```
Comment (albanD) [inline docs/source/conf.py:308]: Adding new entries in this file is not ok.
Please make sure that all the new public APIs that you add are properly documented and show up on the website.

**Review by albanD — CHANGES_REQUESTED** [review]

File: docs/source/conf.py (hunk 2 above)
Comment (Gasoonjia) [reply docs/source/conf.py:308]: Thanks for the feedback. Any example I can use for documentation? I noticed that all public APIs under `torch.ao.quantization.pt2e.graph_utils` are in this file and (looks like) there's not any doc for `graph_utils`.

Comment (Gasoonjia) [reply docs/source/conf.py:308]: If allowed, we can first add `bfs_trace_with_node_process` here, and have another PR for all doc stuff under `graph_utils`.

Comment (jerryzh168) [reply docs/source/conf.py:308]: graph_utils shouldn't be public API, I'd recommend to make this private (add `_`)

**Pull request closed without merging**